### PR TITLE
doc(lsp): various small fixes

### PR DESCRIPTION
**Pull request by DodoMomo**
- remove incorrect usage of docstrings
- fix make_formatting_params return type documentation to 'DocumentFormattingParams'
- make progress_handler private
- fix links

### CODE REVIEW

**Review by mfussenegger — APPROVED** [review]

**Comment by mjlbach** [comment]
A lot of this is auto-generated, so it will get overwritten.

**Comment by mfussenegger** [comment]
> A lot of this is auto-generated, so it will get overwritten.

I think the `See` changes here fixed the docgen hiccup we saw in https://github.com/neovim/neovim/pull/15056/files

File: runtime/doc/lsp.txt [hunk 1]
```diff
@@ -1630,23 +1630,29 @@ save({lenses}, {bufnr}, {client_id})                 *vim.lsp.codelens.save()*
 ==============================================================================
 Lua module: vim.lsp.handlers                                    *lsp-handlers*
 
-                                         *vim.lsp.handlers.progress_handler()*
-progress_handler({_}, {_}, {params}, {client_id})
-                See also: ~
-                    https://microsoft.github.io/language-server-protocol/specifications/specification-current/#workspace_executeCommand
+                                                    *vim.lsp.handlers.hover()*
+hover({_}, {method}, {result}, {_}, {_}, {config})
+                |lsp-handler| for the method "textDocument/hover" >
```
Comment (mjlbach) [inline runtime/doc/lsp.txt:1635]: This sentence is incomplete

File: runtime/doc/lsp.txt [hunk 2]
```diff
@@ -1630,23 +1630,29 @@ save({lenses}, {bufnr}, {client_id})                 *vim.lsp.codelens.save()*
...
                                 • Add borders to the floating window
                                 • See |vim.api.nvim_open_win()|
 
-                See also: ~
-                    https://microsoft.github.io/language-server-protocol/specifications/specification-current/#textDocument_declaration@seehttps://microsoft.github.io/language-server-protocol/specifications/specification-current/#textDocument_definition@seehttps://microsoft.github.io/language-server-protocol/specifications/specification-current/#textDocument_typeDefinition@seehttps://microsoft.github.io/language-server-protocol/specifications/specification-current/#textDocument_implementation|lsp-handler| for the method "textDocument/signatureHelp"
-                    The active parameter is highlighted with
-                    |hl-LspSignatureActiveParameter|>
+                                           *vim.lsp.handlers.signature_help()*
+signature_help({_}, {method}, {result}, {client_id}, {bufnr}, {config})
+                |lsp-handler| for the method "textDocument/signatureHelp". The
```
Comment (mjlbach) [inline runtime/doc/lsp.txt:1653]: incomplete sentence

**Pull request merged**